Update padding for desktop menu dropdown

KaichenWang opened the pull request:
### **Why are these changes introduced?**

<details>
 <summary>Update spacing to values in Figma</summary>

https://www.figma.com/file/JcYdswrk2JhqBTGBVWPM7J/Dawn-2.0?node-id=14848%3A88584
![Screen Shot 2022-02-01 at 2 23 51 PM](https://user-images.githubusercontent.com/3710548/152039514-ad13d5f5-6d3c-46b9-9f98-4d2b1031d666.png)

</details>

### **What approach did you take?**

- Adjust padding values
- Remove additional `padding-right` reserved for caret. Caret is no longer `position: absolute`. When present, caret will push label to the left.

### **Other considerations**

### **Testing steps/scenarios**
- [ ] Visual testing of desktop menu dropdown and its contents
- [ ] Items inside dropdown menu that have children: pay particular attention to caret

### **Demo links**

- [Store](https://os2-demo.myshopify.com/?preview_theme_id=127533547542)
- [Editor](https://os2-demo.myshopify.com/admin/themes/127533547542/editor)

### **Checklist**
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

kmeleta reviewed (approved):
LGTM 👍  [This](https://screenshot.click/02-44-i5lk8-knzza.png) still isn't ideal, but it's certainly no worse than it is currently.

KaichenWang commented:
> LGTM 👍 [This](https://screenshot.click/02-44-i5lk8-knzza.png) still isn't ideal, but it's certainly no worse than it is currently.

Good catch, I think this falls under the ["Incompatibility with long strings" issue](https://github.com/Shopify/dawn/issues/1273), so I've added this as a note there.

tyleralsbury reviewed (approved):
Looks good - just curious, I noticed that you made the caret within the submenu no longer `position: absolute` but the top level menu it seems that it still is.  Is there a particular reason why you changed one but not the other?  It looks like it could use the same treatment.

duygukalaycioglu reviewed (commented):
Overall looking great! When I click "Shop all" the "Shoes" nav link is highlighted with underline as well, is this a bug? 

<img width="590" alt="Screen Shot 2022-02-04 at 10 09 57 AM" src="https://user-images.githubusercontent.com/88160784/152580857-daffe682-bb5c-4004-bcbe-883c0089f5cd.png">

KaichenWang commented:
> Overall looking great! When I click "Shop all" the "Shoes" nav link is highlighted with underline as well, is this a bug?

Good catch. This was due to me messing around with the navigation in the admin. I had added a "All products" link pointing to `/collections/all` under "Shoes", so the underline is technically correct. I reverted that change in the admin and this no longer happens.

![Screen Shot 2022-02-04 at 1 26 15 PM 2](https://user-images.githubusercontent.com/3710548/152583733-e39abc05-6b60-4c04-b9f0-dc860e6db3a3.png)

duygukalaycioglu reviewed (approved):
Thanks for fixing that Kai, looking great!
